Sources/Attribution/AttributionPoster.swift

```diff
@@ -134,22 +135,44 @@ class AttributionPoster {
         postponedAttributionData = postponedData
     }
 
+    private func post(adServicesToken: String) {
+        let currentAppUserID = self.currentUserProvider.currentAppUserID
+
+        // set the cache in advance to avoid multiple post calls
+        let latestNetworkIdsAndAdvertisingIdsSentByNetwork =
+            self.deviceCache.latestAdvertisingIdsByNetworkSent(appUserID: currentAppUserID)
+        var newDictToCache = latestNetworkIdsAndAdvertisingIdsSentByNetwork
```

Comment:
`latestNetworkIdsAndAdvertisingIdsSentByNetwork` seems used only to make a copy of it. You could define only one variable:
```suggestion
        var newDictToCache = self.deviceCache.latestAdvertisingIdsByNetworkSent(appUserID: currentAppUserID)
```